db/migrations/20201220213114_create_circle_basic_schema.js

```diff
@@ -0,0 +1,217 @@
+    t.bigInteger('cover').unsigned()
+    t.bigInteger('avatar').unsigned()
+    t.enu('state', ['active', 'banned', 'archived'])
+      .notNullable()
+      .defaultTo('active')
+    t.bigInteger('owner').unsigned().notNullable()
+    t.string('display_name').notNullable()
+    t.text('description')
+    t.enu('provider', ['stripe']).notNullable().defaultTo('stripe')
+    t.string('provider_product_id').notNullable().unique()
+    t.timestamp('created_at').defaultTo(knex.fn.now())
+    t.timestamp('updated_at').defaultTo(knex.fn.now())
+
+    // foreign keys
+    t.foreign('owner')
+      .references('id')
+      .inTable('user')
+      .onUpdate('CASCADE')
+      .onDelete('CASCADE')
+    t.foreign('cover')
+      .references('id')
+      .inTable('asset')
+      .onUpdate('CASCADE')
+      .onDelete('RESTRICT')
+    t.foreign('avatar')
+      .references('id')
+      .inTable('asset')
+      .onUpdate('CASCADE')
+      .onDelete('RESTRICT')
+  })
+
+  // table for actions on circle, e.g. follow
+  table = 'action_circle'
+
+  await knex('entity_type').insert({ table })
+  await knex.schema.createTable(table, (t) => {
+    t.bigIncrements('id')
+    t.enu('action', ['follow']).defaultTo('follow')
+    t.bigInteger('user_id').unsigned().notNullable()
+    t.bigInteger('target_id').unsigned().notNullable()
+    t.timestamp('created_at').defaultTo(knex.fn.now())
+    t.timestamp('updated_at').defaultTo(knex.fn.now())
+
+    // foreign keys
+    t.foreign('user_id')
+      .references('id')
+      .inTable('user')
+      .onUpdate('CASCADE')
+      .onDelete('CASCADE')
+    t.foreign('target_id')
+      .references('id')
+      .inTable('circle')
+      .onUpdate('CASCADE')
+      .onDelete('CASCADE')
+  })
+
+  // article_circle table
+  table = 'circle_price'
+
+  await knex('entity_type').insert({ table })
+  await knex.schema.createTable(table, (t) => {
+    t.bigIncrements('id')
+
+    t.string('name').notNullable().unique()
+    t.integer('amount').notNullable()
+    t.enu('state', ['active', 'banned', 'archived'])
+    t.enu('currency', ['HKD', 'LIKE']).notNullable().defaultTo('HKD')
+    t.bigInteger('circle_id').unsigned().notNullable()
+
+    t.enu('provider', ['stripe']).notNullable().defaultTo('stripe')
+    t.string('provider_price_id').notNullable().unique()
+    t.timestamp('created_at').defaultTo(knex.fn.now())
+    t.timestamp('updated_at').defaultTo(knex.fn.now())
+
+    // foreign keys
+    t.foreign('circle_id')
+      .references('id')
+      .inTable('circle')
+      .onUpdate('CASCADE')
+      .onDelete('CASCADE')
+  })
+
+  // table for article - circle relationship
+  table = 'article_circle'
+
+  await knex('entity_type').insert({ table })
+  await knex.schema.createTable(table, (t) => {
+    t.bigIncrements('id')
+    t.bigInteger('article_id').unsigned().notNullable()
+    t.bigInteger('circle_id').unsigned().notNullable()
+    t.bigInteger('creator').unsigned().notNullable()
```

Comment:
what does `creator` stands for?

Reply:
Similar with `article_tag`, we can record who add the article to this circle, since we might enable collaboration in the future. But it's currently not needed, removing it for now.